src/NSaga/SqlServer/Common/IQueryWrapper.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using PetaPoco;
+
+namespace NSaga
+{
+    /// <summary>
+    /// Wrapper to be able to delegate queries depending on the type of database.
+    /// </summary>
+    internal interface IQueryWrapper
```

Comment:
Just no. Don't need that. You have 2 IDENTICAL implmenetations of this interface. The only difference are table names. 
And just for that you add a boat-load of abstractions and other code? there are better ways to do this.